mkosi/run.py

```diff
@@ -193,7 +193,7 @@ def run(
     check: bool = True,
     stdout: _FILE = None,
     stderr: _FILE = None,
-    env: Optional[Mapping[str, Any]] = None,
+    env: Mapping[str, Any] = {},
```

Comment:
This should probably not be changed, because that's a footgun. Function arguments are only ever initialised once, so subsequent calls to this function might get previous invocations' `env`. I think the problem is sidestepped in this case, because `env` is overwritten wholesale and the initial `env` not changed, but this is too brittle.

Also, since we pass `env` to a subprocess, can't we whittle this down to `Mapping[str, str]`?

Reply:
We use a read-only type so this isn't a problem